src/main/java/edu/kit/compiler/semantic/SymbolTable.java

```diff
@@ -0,0 +1,143 @@
+package edu.kit.compiler.semantic;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+import java.util.ArrayDeque;
+import java.util.LinkedHashMap;
+
+@Data
+public class SymbolTable  {
+
+    private final ArrayDeque<Change> changes = new ArrayDeque<>();
+    private final SymbolStringTable symbolStringTable = new SymbolStringTable();
+    private Scope currentScope;
+
+    @RequiredArgsConstructor
+    static class Symbol {
+        @NonNull
+        int name;
```

Comment:
`@NonNull` is unnecessary on primitive type variables.

Reply:
This is used to let lombok generate the `@RequiredArgsConstructor`

Reply:
This leaves a warning in my IDE, couldn't we make `name` final?